cpp/src/arrow/compute/exec/exec_plan.h

```diff
@@ -221,6 +228,18 @@ class ARROW_EXPORT ExecNode {
 
   std::string ToString() const;
 
+  /// \brief Is an executor available?
+  bool has_executor() { return plan()->exec_context()->executor() != nullptr; }
```

Comment:
I think all three new methods should be protected (also has_executor can be const, no?)

Reply:
For sure